### ISSUE TITLE
Problems with PBC and units

### ISSUE THREAD

**Issue by pwrobel5** [atomistic-machine-learning/schnetpack#400]
Hello,

Previously I successfully used schnetpack for training models on energies, forces an dipole moments and performing MD simulations for small non-periodic system. Now I am trying to use it for periodic system consisting of 500 atoms. I created extended .xyz file with included lattice vector and used it to create database with either AseEnvironmentProvider or TorchEnvironmentProvider. The problem begins with training NN - AseEnvironmentProvider results in extremely large training times (about 1 day/epoch), while TorchEnvironmentProvider seems to be faster, however is completely unusuable - it produces NaNs in loss function since first training epoch. The problem seems to be related with PBC, since training with standard environment provider has an acceptable speed.

Is the system too big to use PBC? I am also not sure if I use correct units - every position, lattice vectors and cutoff are defined in Angstroms.

I include sample snapshot from extxyz file and script used for training.
[ext_sample.txt](https://github.com/atomistic-machine-learning/schnetpack/files/8543183/ext_sample.txt)

```python
import os

import schnetpack as spk
import schnetpack.train as trn
import torch
from schnetpack.datasets import AtomsData
from schnetpack.environment import AseEnvironmentProvider
from torch.optim import Adam

example = "./ec"
if not os.path.exists(example):
    os.makedirs(example)

emim_fsi_data = AtomsData("ec-01-ase.db", environment_provider=AseEnvironmentProvider(7.5))

train, val, test = spk.train_test_split(
    data=emim_fsi_data,
    num_train=22000,
    num_val=3000,
    split_file=os.path.join(example, "split.npz"),
)

train_loader = spk.AtomsLoader(train, batch_size=2, shuffle=True)
val_loader = spk.AtomsLoader(val, batch_size=2)

means, stddevs = train_loader.get_statistics(spk.Properties.energy, divide_by_atoms=True)

n_features = 128

schnet = spk.representation.SchNet(
    n_atom_basis=n_features,
    n_filters=n_features,
    n_gaussians=25,
    n_interactions=3,
    cutoff=5.,
    cutoff_network=spk.nn.CosineCutoff
)

energy_model = spk.atomistic.Atomwise(
    n_in=n_features,
    property=spk.Properties.energy,
    mean=means[spk.Properties.energy],
    stddev=stddevs[spk.Properties.energy],
    derivative=spk.Properties.forces,
    negative_dr=True
)

model = spk.AtomisticModel(representation=schnet, output_modules=energy_model)

rho_tradeoff = 0.1


def loss(batch, result):
    diff_energy = batch[spk.Properties.energy] - result[spk.Properties.energy]
    err_sq_energy = torch.mean(diff_energy ** 2)

    diff_forces = batch[spk.Properties.forces] - result[spk.Properties.forces]
    err_sq_forces = torch.mean(diff_forces ** 2)

    err_sq = rho_tradeoff * err_sq_energy + (1 - rho_tradeoff) * err_sq_forces
    return err_sq


optimizer = Adam(model.parameters(), lr=5e-4)

metrics = [
    spk.metrics.MeanAbsoluteError(spk.Properties.energy),
    spk.metrics.MeanAbsoluteError(spk.Properties.forces)
]

hooks = [
    trn.CSVHook(log_path=example, metrics=metrics),
    trn.ReduceLROnPlateauHook(
        optimizer,
        patience=5, factor=0.8, min_lr=1e-6,
        stop_after_min=True
    )
]

trainer = trn.Trainer(
    model_path=example,
    model=model,
    hooks=hooks,
    loss_fn=loss,
    optimizer=optimizer,
    train_loader=train_loader,
    validation_loader=val_loader
)

if torch.cuda.is_available():
    device = "cuda"
else:
    device = "cpu"

n_epochs = 200
trainer.train(device=device, n_epochs=n_epochs)
```

Cheers,
Piotr

**Comment by ktschuett** [comment]
Yes, we notice that the `TorchEnvironmentProvider` can be problematic in some circumstances. Make sure that you are using a current version of ASE, since the improved their neighbor list at some point. I noticed you are using an internal cutoff of 5, but a cutoff of 7.5 in the environment provider, which wastes some computation.

You could also try the new schnetpack located in the `dev` branch (albeit with breaking API changes): we have implemented a `CachedNeighborList` that can wrap e.g. ASE and cache the results dynamically on the disk for future epochs.

**Comment by pwrobel5** [comment]
I have been using the most recent version of ASE, so it is not the case here.

I am trying to use the new version in the ```dev``` branch, however I have a problem - since the last update of example script on which I am basing my code (https://github.com/atomistic-machine-learning/schnetpack/blob/dev/src/examples/ethanol_example.py) some changes in ```AtomisticModel``` class have been made. Should I use other class instead of it in line 90? The new version has no ```datamodule``` initial argument. May I get a minimal working example of the new API usage?
